src/models/politician.ts

```diff
@@ -37,3 +37,19 @@ export type PartyRoleHistory = {
 	startedAt: Date;
 	endedAt?: Date;
 };
+
+export type RoleHistory = {
+	role: string;
+	startedAt: Date;
+	endedAt?: Date;
+};
+
+export function getRoleHistoryAtTime<T extends RoleHistory>(
+	histories: T[],
+	time: Date
+): T | undefined {
+	return histories.find(
```

Comment:
I'm curious do we have a use case for making generatic type `T` instead of just use `RoleHistory` for histories directly?

Reply:
In that scenario, the return type of the function will be `RoleHistory` which will be unusable for its use case since we want to get other properties besides what declared in the `RoleHistory`.

For example:
```
getRoleHistoryAtTime(bill.proposedLedByPolitician.partyRoles, bill.proposedOn)?.party.name // Won't be able to extract `party` for this function
```